rossum_api/elis_api_client.py

```diff
@@ -316,6 +317,31 @@ async def poll_annotation_until_imported(
             annotation_id, lambda a: a.status not in ("importing", "created"), **poll_kwargs
         )
 
+    async def poll_task(
+        self,
+        resource_id: int,
+        predicate: Callable[[Task], bool],
+        sleep_s: int = 3,
+        sideloads: Sequence[str] = (),
+    ) -> Task:
+        """Poll on Task until predicate is true.
+
+        Sideloading is done only once after the predicate becomes true to avoid spamming the server.
+        """
+        task_json = await self._http_client.fetch_one(Resource.Task, resource_id)
+        # Parse early, we want predicate to work with Annotation instances for convenience
+        task = self._deserializer(Resource.Task, task_json)
+
+        while not predicate(task):
+            await asyncio.sleep(sleep_s)
+            task_json = await self._http_client.fetch_one(Resource.Task, resource_id)
+            task = self._deserializer(Resource.Task, task_json)
+
+        if sideloads:
+            await self._sideload(task_json, sideloads)
+        return self._deserializer(Resource.Task, task_json)
+
+
```

Comment:
That was my thought tool, I was making "poll_resource". It ended up being a huge pain because of I believe the deserialization, sth with the Resource enum? There was definitely some hidden betrayal. 
Right, I think we would need to add a parent class for the models that it makes sense to poll and have the appropriate structure for it.

It was early in the development but in the end I argued that if Tasks stay, this can be turned into generic polling function but once-duplicate is fine.